core/running/test-runner.ts

```diff
@@ -44,10 +45,19 @@ export class TestRunner {
        await this._runTests(testSetRunInfo, testSetResults);
     }
 
-    private async _runTests(testSetRunInfo: TestSetRunInfo, results: TestSetResults) {
+    /**
+     * Defined the call back function to be called when the test is completed
+     */
+    public onTestComplete( testCompleteCB: IOnTestCompleteCBFunction) {
```

Comment:
working on this, do we need to release these CB? If yes, how? at the end of the run method?

If we do it at the end of run, the sequence feel strange.
* create testRunner
* link CB
* link CB
* tr.Run() => event raised, CB released
* tr.Run() => no event raised

:-( 

Should we pass the CB as parameters of the run method?